bot/exts/moderation/incidents.py

```diff
@@ -402,10 +486,147 @@ async def on_raw_reaction_add(self, payload: discord.RawReactionActionEvent) ->
+
+            if webhook_msg_id:
+                await self.incidents_webhook.edit_message(
+                    message_id=webhook_msg_id,
+                    embeds=[embed for embed in webhook_embed_list if embed is not None],
+                )
+                return
+
+            await self.send_message_link_embeds(webhook_embed_list, msg_after, self.incidents_webhook)
+
+    @Cog.listener()
+    async def on_raw_message_delete(self, payload: discord.RawMessageDeleteEvent) -> None:
+        """
+        Delete message link embeds for `payload.message_id`.
+
+        Search through the cache for message, if found delete it from cache and channel.
+        """
+        await self.delete_msg_link_embed(payload.message_id)
+
+    async def extract_message_links(self, message: discord.Message) -> t.Optional[t.List[discord.Embed]]:
+        """
+        Checks if there's any message links in the text content.
+
+        Then passes the the message_link into `make_message_link_embed` to format a
+        embed for it containing information about the link.
+
+        As discord only allows a max of 10 embeds in a single webhook, just send the
+        first 10 embeds and don't care about the rest.
+
+        If no links are found for the message, it logs a trace statement.
+        """
+        message_links = DISCORD_MESSAGE_LINK_RE.findall(str(message.content))
+
+        if message_links:
+            embeds = []
+            for message_link in message_links[:10]:
+                ctx = await self.bot.get_context(message)
+                embed = await make_message_link_embed(ctx, message_link[0])
+                if embed:
+                    embeds.append(embed)
+
+            return embeds
+
+        log.trace(
+            f"Skipping discord message link detection on {message.id}: message doesn't qualify."
+        )
+
+    async def send_message_link_embeds(
+            self,
+            webhook_embed_list: t.List,
+            message: discord.Message,
+            webhook: discord.Webhook,
+    ) -> t.Optional[int]:
+        """
+        Send Message Link Embeds to #incidents channel.
+
+        Uses the `webhook` passed in as a parameter to send
+        the embeds in the `webhook_embed_list` parameter.
+
+        After sending each webhook it maps the `message.id`
+        to the `webhook_msg_ids` IDs in the async redis-cache.
+        """
+        try:
+            webhook_msg = await webhook.send(
+                embeds=[embed for embed in webhook_embed_list if embed is not None],
+                username=sub_clyde(message.author.name),
+                avatar_url=message.author.avatar_url,
+                wait=True,
+            )
+
+        except discord.DiscordException:
+            log.exception(
+                f"Failed to send message link embed {message.id} to #incidents."
+            )
+
+        else:
+            await self.message_link_embeds_cache.set(int(message.id), int(webhook_msg.id))
```

Comment:
I don't think these need int() casts.. do they?

Reply:
This was added as it was having the same issue as [this](https://github.com/python-discord/bot/pull/1446#discussion_r631685982).

Reply:
Try not adjusting code if there are tests failing and you can't figure out how to reproduce the test failure with manual code. `message.id` should never be non-int, so if you're having to cast there you're having other problems (such as with the test). Otherwise the tests become part of your code, and are not really suitable as tests any longer. 

This issue is because the Mock being used isn't having .id set appropriately. Try using `breakpoint()` or another debugger when this errors to figure out exactly what is being passed here during your tests.